kazoo/protocol/connection.py

```diff
@@ -643,7 +631,7 @@ def _connect(self, host, port):
 
         with self._socket_error_handling():
             self._socket = self.handler.create_connection(
-                address=(host, port),
+                address=(hostip, port),
```

Comment:
Did you notice issues when using the previous behaviour?

Reply:
This was early on. If I remember properly, without this, I was seeing IPs passed to the SASL library which was then creating principals in the form `zookeeper/1.2.3.4@myrealm` instead of `zookeeper/myhostname@myrealm`.

Do you have a concern with this?
As I said, this was a while ago, I can double check...

Reply:
Not much concern, just wanting to understand the reason for the change and keep some history about it.